assets/js/ReorderColumns.test.js

```diff
@@ -0,0 +1,194 @@
+    const FILTER_WFM_ID = 50;
+
+    const REORDER_WFM_ID = 85;
+    const REORDER_HISTORY_PAR_ID = 90;
+
+    const NEW_REORDER_WFM_ID = 135;
+    const NEW_REORDER_HISTORY_PAR_ID = 105;
+
+    const REORDER_MODULE_ID = 24;
+    const WF_ID = 10;
+
+    var initialState = {
+        reorderModuleId: REORDER_MODULE_ID,
+        workflow: {
+            id: WF_ID,
+            wf_modules: [
+                {
+                    id: LOADURL_WFM_ID,
+                    module_version: {
+                        module: {
+                            id_name: 'loadurl'
+                        }
+                    }
+                },
+                {
+                    id: FILTER_WFM_ID,
+                    module_version: {
+                        module: {
+                            id_name: 'filter'
+                        }
+                    }
+                },
+                {
+                    id: REORDER_WFM_ID,
+                    module_version: {
+                        module: {
+                            id_name: 'reorder-columns'
+                        },
+                    },
+                    parameter_vals: [
+                        {
+                            id: REORDER_HISTORY_PAR_ID,
+                            parameter_spec: {
+                                id_name: 'reorder-history'
+                            },
+                            value: JSON.stringify([{
+                                column: 'existing_test_col',
+                                from: 2,
+                                to: 4
+                            }])
+                        }
+                    ]
+                }
+            ]
+        }
+    };
+
+    const addModuleResponse = {
+        id: NEW_REORDER_WFM_ID,
+        module_version: {
+            module: {
+                id_name: 'reorder-columns'
+            }
+        },
+        parameter_vals: [
+            {
+                id: NEW_REORDER_HISTORY_PAR_ID,
+                parameter_spec: {
+                    id_name: 'reorder-history'
+                },
+                value: ''
+            }
+        ]
+    };
+
+    beforeEach(() => {
+       api = {
+           onParamChanged: jest.fn().mockReturnValue(Promise.resolve()),
+           addModule: jsonResponseMock(addModuleResponse),
+           setSelectedWfModule: jest.fn().mockReturnValue(Promise.resolve())
+       };
+       mockAPI(api);
+       mockStoreApi(api);
+       store = {
+           getState: () => initialState,
+           dispatch: jest.fn().mockReturnValue(Promise.resolve())
+       };
+       mockStore(store);
+    });
+
+    it('Adds a new reorder module after the current non-reorder module '
+        + 'if a reorder module does not exist next to it and sets parameters correctly', (done) => {
+        updateReorder(LOADURL_WFM_ID, {
+            column: 'test_col',
+            from: 3,
+            to: 0
+        });
+        setImmediate(() => {
```

Comment:
I don't think you need the setImmediate here, as you're not waiting for a promise to resolve (nothing here depends on the result from a mocked API call)